unity-renderer/Assets/DCLPlugins/ECS7/Systems/GltfContainerLoadingStateSystem/GltfContainerLoadingStateSystem.cs

```diff
@@ -0,0 +1,59 @@
+using DCL.Controllers;
+using DCL.ECS7;
+using DCL.ECS7.InternalComponents;
+using DCL.ECSComponents;
+using DCL.ECSRuntime;
+using DCL.Models;
+
+namespace ECSSystems.GltfContainerLoadingStateSystem
+{
+    public class GltfContainerLoadingStateSystem
+    {
+        private readonly IInternalECSComponent<InternalGltfContainerLoadingState> gltfContainerLoadingStateComponent;
+        private readonly IECSComponentWriter componentWriter;
+
+        public GltfContainerLoadingStateSystem(IECSComponentWriter componentWriter,
+            IInternalECSComponent<InternalGltfContainerLoadingState> gltfContainerLoadingStateComponent)
+        {
+            this.componentWriter = componentWriter;
+            this.gltfContainerLoadingStateComponent = gltfContainerLoadingStateComponent;
+        }
+
+        public void Update()
+        {
+            var components = gltfContainerLoadingStateComponent.GetForAll();
```

Comment:
Currently this implementation traverses all the entities with `IInternalECSComponent<InternalGltfContainerLoadingState>`, so basically EVERY entity with a GLTF (since we are adding that internal component to all).
Wouldn't it be better to just traverse the entities that have the new (**not internal**) [GltfContainerLoadingState](https://github.com/decentraland/unity-renderer/blob/781b320216e8aa4aa61bc04c4dfcc98641eab6bc/unity-renderer/Assets/Scripts/MainScripts/DCL/DecentralandProtocol/GltfContainerLoadingState.gen.cs) component instead? since those would be the only ones that we are interested in updating and reporting its loading state...

Maybe I'm missing the point here as I thought that to use the new`GltfContainerLoadingState` component from a scene with the SDK it should be added from the scene to an entity and then read it later... if we are adding that data to EVERY entity with GltfContainer why not add and update that loading state property in the GltfContainer component itself???

Reply:
> Wouldn't it be better to just traverse the entities that have the new (not internal) [GltfContainerLoadingState](https://github.com/decentraland/unity-renderer/blob/781b320216e8aa4aa61bc04c4dfcc98641eab6bc/unity-renderer/Assets/Scripts/MainScripts/DCL/DecentralandProtocol/GltfContainerLoadingState.gen.cs) component instead?

No, since scene won't be sending the `GltfContainerLoadingState` to the renderer, the renderer will be sending the `GltfContainerLoadingState` to the scene. It's not sent by request, every `GltfContainer` expect to receive a `GltfContainerLoadingState`

> Maybe I'm missing the point here as I thought that to use the newGltfContainerLoadingState component from a scene with the SDK it should be added from the scene to an entity and then read it later... if we are adding that data to EVERY entity with GltfContainer why not add and update that loading state property in the GltfContainer component itself??? and avoid having a new unneeded component.

`GltfContainer` is sent from the scene to the renderer, not the other way around
`GltfContainerLoadingState` is sent from the renderer to the scene, not the other way around
so we need two different components to avoid any write conflict.
plus, they are related but conceptually different components, one contains data of the gltf and the other the data of it loading state